### DISCUSSION TITLE
Nativescript Getters

### DISCUSSION THREAD

**Discussion by darko-bw** [NativeScript/rfcs#14] [1. Ideas & Discussions]
Hi guys, thank you for your time and effort to keep Nativescript up and running.

I am using a lot nativescript-getters plugin to get views by tag, class names, values etc etc.

Do you think we can get it integrated into NS 8? Or maybe we can have anything similar to it?

Many thanks.
Edit: Relevant Link: https://plugins.nativescript.rocks/plugin/nativescript-getters

**Comment by NathanaelA** [comment]
Very interesting, my NS-Dom plugin (https://plugins.nativescript.rocks/plugin/nativescript-dom) was written primarily to do this -- this really only plays into NS without a flavor as most the other flavors have their own built in methods to grab things.   

I have a couple questions to figure out what pieces we might want to add:
I'm personally for adding:
- getElementsByClass   (ns-dom)  - GetViewsByClass (ns-getter)
- getElementsByTagName  (ns-dom) - getViewsByTag (ns-getter)
- querySelector (doesn't exist in any plugin that I'm aware of)

@basil-wolf - Are you using any of the other helpers in the NS-getter library?    I looked at the list and couldn't think of any real uses that those additional helpers would serve in core, but if you have some use cases for any of the other getters that would help.     I know in my plugin I added the `runAgainst*` functions as a helper, but I don't think that they should be added to core, I think adding the three above helpers would probably beef up the support to support pretty much everything.

**Reply by darko-bw** [reply]
Hi @NathanaelA 
Thank you. I agree with you on adding those above. I am also using:

- getViewsByProperty (Aliases: Prop, Attribute or Attr)
- getViewsByValuePair (getViewsByValuePair("checked", true))

They are very useful in many cases. And I think your **runAgainst*** would benefit the NS as a big competitor. It will attract a lot of web developers to look at NS as a leader on the market.

**Reply by rigor789** [reply]
These could be covered by `querySelector` if we were to implement support: 
```js
querySelector('[checked=true]');
```

**Reply by darko-bw** [reply]
@rigor789 sure, I understand. I think the main idea is to minimise the dependence on plugins for such an essential functionality.

**Reply by rigor789** [reply]
Right - I was just saying that @NathanaelA's list of the 3 methods would include `getViewsByProperty` and `getViewsByValuePair` through `querySelector`.

**Reply by darko-bw** [reply]
@rigor789 and @NathanaelA
Would be great to have it! 
Thank you again for all your hard work. You make a massive difference and keep NS truly versatile.

**Reply by vallemar** [reply]
I am more a front and back developer than a mobile developer and if I saw `querySelector` in NS it would be great since for frontend I think it is the most used. I also agree with the others but I think that with `querySelector` all are covered, it is the most versatile
